File: 724-find-pivot-index/find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        // int totalSum = 0;
        // int leftSum = 0;
        // int n = nums.length;
        
        // // Step 1: Poore array ka total sum nikal lo (Right Side tarazu)
        // for (int i = 0; i < n; i++) {
        //     totalSum += nums[i];
        // }
        
        // // Step 2: Har ek index par jaakar check karo
        // for (int i = 0; i < n; i++) {
        //     // Right side se current element ko minus karo
        //     totalSum -= nums[i]; 
            
        //     // Agar left aur right side ka sum barabar ho gaya
        //     if (leftSum == totalSum) {
        //         return i; // Mil gaya pivot index!
        //     }
            
        //     // Agar barabar nahi hai, toh ise left side mein plus kar do
        //     leftSum += nums[i];
        // }
        // return -1;

        // for each loop ke mada se 
        int totalSum = 0;
        int leftSum = 0;
        
        // Step 1: Poore array ka total sum nikal lo
        for (int num : nums) {
            totalSum += num;
        }
        
        // Aapka bataya hua variable 'i' index track karne ke liye
        int i = 0; 
        
        // Step 2: Tarazu wala logic check karo
        for (int num : nums) {
            totalSum -= num; // Right side se current element hatao
            
            // Agar dono side barabar hain toh index 'i' return karo
            if (leftSum == totalSum) {
                return i; 
            }
            
            leftSum += num; // Current element ko left side mein jodo
            i++;             // Loop ki speed ke sath match karne ke liye sirf ++ kiya
        }
        
        return -1;
    }

}

File: 724-find-pivot-index/find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int total_sum = 0,left_sum = 0;
        for(int num : nums){
            total_sum += num;
        }
        int i = 0;
        for(int num : nums){
            total_sum -= num;

            if(left_sum == total_sum){
                return i;
            }
            left_sum += num;
            i++;
        }
        return -1;



        // simple loop wala
        // int total_sum = 0;
        // int left_sum = 0;
        // int n = nums.size();
        
        // // Step 1: Poore array ka total sum nikal lo (Right Side tarazu)
        // for (int i = 0; i < n; i++) {
        //     total_sum += nums[i];
        // }
        
        // // Step 2: Har ek index par jaakar check karo
        // for (int i = 0; i < n; i++) {
        //     // Right side se current element ko minus karo
        //     total_sum -= nums[i]; 
            
        //     // Agar left aur right side ka sum barabar ho gaya
        //     if (left_sum == total_sum) {
        //         return i; // Mil gaya pivot index!
        //     }
            
        //     // Agar barabar nahi hai, toh ise left side mein plus kar do
        //     left_sum += nums[i];
        // }
        
        // // Agar poore loop mein kahin nahi mila toh -1 return karo
        // return -1; 
    }
};